### PR TITLE
Fix IMU failover (simpler alternative)

### PR DESCRIPTION
**Pull request by tridge**
This is a simpler version of #11139 which involves no EKF changes
I have test flown this on a copter and there is no change in attitude on IMU failure

### CODE REVIEW

**Comment by priseborough** [comment]
Have you performed the same test without the patch? Following the use of _gyro_estimate I don't see how this changes the IMU instance used by the EKF for navigation.

**Comment by priseborough** [comment]
OK, have answered my own question. It addresses the loss of control issue. Estimation should be using the correct IMU..

File: libraries/AP_AHRS/AP_AHRS_NavEKF.cpp [hunk 1]
```diff
@@ -188,7 +188,7 @@ void AP_AHRS_NavEKF::update_EKF2(void)
 
             // calculate corrected gyro estimate for get_gyro()
             _gyro_estimate.zero();
-            if (primary_imu == -1) {
+            if (primary_imu == -1 || !_ins.get_gyro_health(primary_imu)) {
                 // the primary IMU is undefined so use an uncorrected default value from the INS library
```
Comment (peterbarker) [inline libraries/AP_AHRS/AP_AHRS_NavEKF.cpp]: Comment is now out of date.

File: libraries/AP_AHRS/AP_AHRS_NavEKF.cpp [hunk 2]
```diff
@@ -261,7 +261,7 @@ void AP_AHRS_NavEKF::update_EKF3(void)
...
```
Comment (peterbarker) [inline libraries/AP_AHRS/AP_AHRS_NavEKF.cpp]: Comment is now out of date

File: libraries/AP_InertialSensor/AP_InertialSensor.h [hunk 3]
```diff
@@ -274,6 +274,14 @@ class AP_InertialSensor : AP_AccelCal_Client
     // return time in microseconds of last update() call
     uint32_t get_last_update_usec(void) const { return _last_update_usec; }
 
+    void kill_imu(uint8_t imu_idx, bool kill_it) {
```
Comment (peterbarker) [inline libraries/AP_InertialSensor/AP_InertialSensor.h]: `set_imu_life_status`?

File: libraries/AP_InertialSensor/AP_InertialSensor_SITL.cpp [hunk 4]
```diff
@@ -179,15 +179,19 @@ void AP_InertialSensor_SITL::timer_update(void)
 #endif
     for (uint8_t i=0; i<INS_SITL_INSTANCES; i++) {
         if (now >= next_accel_sample[i]) {
-            generate_accel(i);
-            while (now >= next_accel_sample[i]) {
-                next_accel_sample[i] += 1000000UL / accel_sample_hz[i];
+            if (((1U<<i) & sitl->accel_fail_mask) == 0) {
```
Comment (peterbarker) [inline libraries/AP_InertialSensor/AP_InertialSensor_SITL.cpp]: This could have been done as a `continue` in place of increasing the nesting.

**Comment by tridge** [comment]
new version of this is in master

**Pull request closed without merging**